File: backend/waynder/src/main/java/com/savadev25/waynder/entity/Plan.java
package com.savadev25.waynder.entity;


import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Entity
@Table(name = "plans")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Plan {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    // A route's plan_id is nullable at the DB level (a route may exist without a plan),
    // this side just lists whichever routes currently point at this plan.
    @OneToMany(mappedBy = "plan")
    @Builder.Default
    private Set<Route> routes = new HashSet<>();

    @PrePersist
    protected void onCreate() {
        createdAt = Instant.now();
    }
}

File: frontend/src/models/Plan.js
/**
 * @typedef {Object} Plan
 * @property {string} id
 * @property {string} userId
 * @property {string} name
 * @property {string|null} description
 * @property {string} createdAt
 */

export function mapPlan(json) {
  return {
    id: json.id,
    userId: json.userId,
    name: json.name,
    description: json.description ?? null,
    createdAt: json.createdAt,
  };
}